File: backend/models/nmc.py
from pydantic import BaseModel


class RebootRequest(BaseModel):
    reason: str = "User request"


class WanModeRequest(BaseModel):
    WanMode: str
    Username: str | None = None
    Password: str | None = None


class LanIPConfig(BaseModel):
    Address: str
    Netmask: str
    DHCPEnable: bool
    DHCPMinAddress: str
    DHCPMaxAddress: str
    LeaseTime: int | None = None


class RemoteAccessCreate(BaseModel):
    username: str | None = None
    password: str | None = None
    port: int | None = None
    timeout: int | None = None
    sourcePrefix: str | None = None
    accessType: str | None = None
    secure: bool | None = None


class RemoteAccessDisable(BaseModel):
    accessType: str | None = None


class IPv6Config(BaseModel):
    Enable: bool
    UserRequested: bool | None = None
    IPv4UserRequested: bool | None = None


class IPTVMultiscreenRequest(BaseModel):
    Enable: bool


class LedConfig(BaseModel):
    state: str
    color: str


class ContainerNetworkConfig(BaseModel):
    Address: str | None = None
    Netmask: str | None = None
    DHCPEnable: bool | None = None
    DHCPMinAddress: str | None = None
    DHCPMaxAddress: str | None = None
    LeaseTime: int | None = None


class NetworkBRRequest(BaseModel):
    state: bool | None = None


class BackupRequest(BaseModel):
    delay: bool | None = None


class WlanTimerRequest(BaseModel):
    Timeout: int

File: frontend/src/app/models/nmc.ts
export interface RebootRequest {
  reason: string; // default: "User request"
}

export interface WanModeRequest {
  WanMode: string;
  Username?: string;
  Password?: string;
}

export interface LanIPConfig {
  Address: string;
  Netmask: string;
  DHCPEnable: boolean;
  DHCPMinAddress: string;
  DHCPMaxAddress: string;
  LeaseTime?: number;
}

export interface RemoteAccessCreate {
  username?: string;
  password?: string;
  port?: number;
  timeout?: number;
  sourcePrefix?: string;
  accessType?: string;
  secure?: boolean;
}

export interface RemoteAccessDisable {
  accessType?: string;
}

export interface IPv6Config {
  Enable: boolean;
  UserRequested?: boolean;
  IPv4UserRequested?: boolean;
}

export interface IPTVMultiscreenRequest {
  Enable: boolean;
}

export interface LedConfig {
  state: string;
  color: string;
}

export interface ContainerNetworkConfig {
  Address?: string;
  Netmask?: string;
  DHCPEnable?: boolean;
  DHCPMinAddress?: string;
  DHCPMaxAddress?: string;
  LeaseTime?: number;
}

export interface NetworkBRRequest {
  state?: boolean;
}

export interface BackupRequest {
  delay?: boolean;
}

export interface WlanTimerRequest {
  Timeout: number;
}
